commands/audit/jas/iac/iacscanner.go

```diff
@@ -30,38 +33,43 @@ type IacScanManager struct {
 // []utils.SourceCodeScanResult: a list of the iac violations that were found.
 // bool: true if the user is entitled to iac scan, false otherwise.
 // error: An error object (if any).
-func RunIacScan(scanner *jas.JasScanner) (results []*sarif.Run, err error) {
-	iacScanManager := newIacScanManager(scanner)
-	log.Info("Running IaC scanning...")
-	if err = iacScanManager.scanner.Run(iacScanManager); err != nil {
+func RunIacScan(auditParallelRunner *utils.AuditParallelRunner, scanner *jas.JasScanner, extendedScanResults *utils.ExtendedScanResults, module jfrogappsconfig.Module, threadId int) (err error) {
+	var scannerTempDir string
+	if scannerTempDir, err = jas.CreateScannerTempDirectory(scanner, string(utils.IaC)); err != nil {
+		return
+	}
+	iacScanManager := newIacScanManager(scanner, scannerTempDir)
+	log.Info("[thread_id: " + strconv.Itoa(threadId) + "] Running IaC scanning...")
+	if err = iacScanManager.scanner.Run(iacScanManager, module); err != nil {
 		err = utils.ParseAnalyzerManagerError(utils.IaC, err)
 		return
 	}
 	if len(iacScanManager.iacScannerResults) > 0 {
-		log.Info("Found", utils.GetResultsLocationCount(iacScanManager.iacScannerResults...), "IaC vulnerabilities")
+		log.Info("[thread_id: "+strconv.Itoa(threadId)+"] Found", utils.GetResultsLocationCount(iacScanManager.iacScannerResults...), "IaC vulnerabilities")
 	}
-	results = iacScanManager.iacScannerResults
+	results := iacScanManager.iacScannerResults
+	auditParallelRunner.Mu.Lock()
+	extendedScanResults.IacScanResults = append(extendedScanResults.IacScanResults, results...)
+	auditParallelRunner.Mu.Unlock()
```

Comment:
this is not related to this logic, use it outside of this and return the result of the scan to be used outside.